presto-spi/src/main/java/com/facebook/presto/spi/connector/ConnectorMetadata.java

```diff
@@ -266,6 +263,25 @@ default void finishDelete(ConnectorSession session, ConnectorTableHandle tableHa
     }
 
     /**
+     * Notifies when select query starts
+     *
+     * @return serializable {@link ConnectorTableHandle}. This handle will be brought back in {@link #finishSelect(ConnectorSession, ConnectorTableHandle)} if null returned, no finishSelect will be called.
+     */
+    default ConnectorTableHandle beginSelect(ConnectorSession session, ConnectorTableHandle tableHandle, Optional<ConnectorTableLayoutHandle> layoutHandle, Collection<ColumnHandle> columnHandles)
```

Comment:
Data reading actually happens in workers using `ConnectorPageSourceProvider`. Resources are typically acquired there as needed. For something like a consistent snapshot that all the readers will use, that should actually happen as part of the transaction, so that it can be shared by all tables used in the transaction (or the same table read multiple times).

We have many connectors and none of them need something like this, nor can I imagine a scenario in which it is needed.

If a connector wants to store additional information, it can do so itself in the transaction (there is a `ConnectorMetadata` instance per transaction).